code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -676,9 +691,12 @@
 	if(!istype(H, /mob/living/carbon/human))
 		return 1
 
-	if(H.mind.special_role == "apprentice")
+	if(H.mind.special_role == "apprentice" && owner_is_apprentice == 0)
```

Comment:
See above comment. You can access the owner's mind and shouldn't need this new var